parcels/field.py

```diff
@@ -250,14 +250,23 @@ def __init__(self, name, data, lon=None, lat=None, depth=None, time=None, grid=N
         if not self.data.dtype == np.float32:
             logger.warning_once("Casting field data to np.float32")
             self.data = self.data.astype(np.float32)
+        # Make a copy of the transposed array to enforce
+        # C-contiguous memory layout for JIT mode.
         if transpose:
-            # Make a copy of the transposed array to enforce
-            # C-contiguous memory layout for JIT mode.
             self.data = np.transpose(self.data).copy()
-        if self.grid.zdim > 1:
-            self.data = self.data.reshape((self.grid.tdim, self.grid.zdim, self.grid.ydim, self.grid.xdim))
         else:
-            self.data = self.data.reshape((self.grid.tdim, self.grid.ydim, self.grid.xdim))
+            self.data = self.data.copy()
```

Comment:
So why do we need to copy() even if we're not transposing?

Reply:
If the user does a transpose in its own code, the same issue occurs...

Reply:
But most people won't transpose their own code. This sounds like a hefty overhead for most usecases. Is there a way to check if a numpy array has been transposed, so that we only do it then?